app/Controller/CoPeopleController.php

```diff
@@ -554,18 +554,35 @@ public function find() {
       
       foreach($people as $p) {
         $label = generateCn($p['Name'][0]);
-        
-        $id = Hash::extract($p['Identifier'], '{n}[type=uid]');
-        
-        if(!empty($id[0]['identifier'])) {
-          $label .= " (" . $id[0]['identifier'] . ")";
+        $idArr = Hash::extract($p['Identifier'], '{n}[type=uid]');
+        $emailArr = $p['EmailAddress'];
+        $email = '';
+        $id = '';
+          
+        // Iterate over the email array
+        if (!empty($emailArr)) {
+          foreach($emailArr as $k => $e) {
+            $email .= $e['mail'];
+            if($k !== array_key_last($emailArr)) {
+              $email .= ', ';
+            } 
+          }
         }
         
+        // Set the identifier for display (and limit it to 30 characters max)
+        if (!empty($idArr[0]['identifier'])) {
+          $id = mb_strimwidth($idArr[0]['identifier'], 0, 30, '...');
+        }
+         
         // Make sure we don't already have an entry for this CO Person ID
         if(!Hash::check($matches, '{n}[value='.$p['CoPerson']['id'].']')) {
           $matches[] = array(
             'value' => $p['CoPerson']['id'],
-            'label' => $label
+            'label' => $label,
+            'email' => $email,
+            'emailLabel' => _txt('fd.email_address.mail'),
```

Comment:
I'm thinking this is wrong for right-to-left languages; we should instead format the whole string via PHP text replacements and then pass that. It's both simpler and avoids i18n issues. Will convert to draft and fix this.